src/compute_budget.zig

```diff
@@ -0,0 +1,130 @@
+const std = @import("std");
+const sol = @import("solana-program-sdk");
+const bincode = @import("bincode");
+
+const Account = sol.Account;
+
+const ComputeBudget = @This();
+
+pub const id = sol.PublicKey.comptimeFromBase58("ComputeBudget111111111111111111111111111111");
+
+pub fn requestHeapFrame(allocator: std.mem.Allocator, bytes: u32) !sol.Instruction {
+    const data = try bincode.writeAlloc(allocator, ComputeBudget.Instruction{
+        .request_heap_frame = .{ .bytes = bytes },
+    }, .{});
+
+    return sol.Instruction.from(.{
+        .program_id = &id,
+        .accounts = &[_]Account.Param{},
+        .data = data,
+    });
+}
+
+pub fn setComputeUnitLimit(allocator: std.mem.Allocator, units: u32) !sol.Instruction {
+    const data = try bincode.writeAlloc(allocator, ComputeBudget.Instruction{
+        .set_compute_unit_limit = .{ .units = units },
+    }, .{});
+
+    return sol.Instruction.from(.{
+        .program_id = &id,
+        .accounts = &[_]Account.Param{},
+        .data = data,
+    });
+}
+
+pub fn setComputeUnitPrice(allocator: std.mem.Allocator, micro_lamports: u64) !sol.Instruction {
+    const data = try bincode.writeAlloc(allocator, ComputeBudget.Instruction{
+        .set_compute_unit_price = .{ .micro_lamports = micro_lamports },
+    }, .{});
+
+    return sol.Instruction.from(.{
+        .program_id = &id,
+        .accounts = &[_]Account.Param{},
+        .data = data,
+    });
+}
+
+pub fn setLoadedAccountsDataSizeLimit(allocator: std.mem.Allocator, bytes: u32) !sol.Instruction {
+    const data = try bincode.writeAlloc(allocator, ComputeBudget.Instruction{
+        .set_loaded_accounts_data_size_limit = .{ .bytes = bytes },
+    }, .{});
+
+    return sol.Instruction.from(.{
+        .program_id = &id,
+        .accounts = &[_]Account.Param{},
+        .data = data,
+    });
+}
+
+pub const Instruction = union(enum(u8)) {
+    // deprecated variant, reserved value.
+    unused,
+    /// Request a specific transaction-wide program heap region size in bytes.
+    /// The value requested must be a multiple of 1024. This new heap region
+    /// size applies to each program executed in the transaction, including all
+    /// calls to CPIs.
+    request_heap_frame: struct {
+        bytes: u32,
+    },
+    /// Set a specific compute unit limit that the transaction is allowed to consume.
+    set_compute_unit_limit: struct {
+        units: u32,
+    },
+    /// Set a compute unit price in "micro-lamports" to pay a higher transaction
+    /// fee for higher transaction prioritization.
+    set_compute_unit_price: struct {
+        micro_lamports: u64,
+    },
+    /// Set a specific transaction-wide account data size limit, in bytes, is allowed to load.
+    set_loaded_accounts_data_size_limit: struct {
+        bytes: u32,
+    },
+};
+
+test "build request heap frame ix" {
+    const ix = try requestHeapFrame(std.testing.allocator, 4000);
+
+    defer std.testing.allocator.free(ix.data[0..ix.data_len]);
```

Comment:
Also, using an external allocator brings some extra steps to clean up the memory.

We can encapsulate cleaning logic inside the `sol.Instruction` and add `deinit` method

Reply:
Yeah that would definitely be a better option. We might even want to think about how to totally avoid allocations for program instructions, since the instruction data and account meta sizes should typically be known at compile time.